library/ssl_tls.c

```diff
@@ -3985,7 +4035,15 @@ void mbedtls_ssl_handshake_free( mbedtls_ssl_context *ssl )
 #if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     psa_pake_abort( &handshake->psa_pake_ctx );
-    psa_destroy_key( handshake->psa_pake_password );
+    /*
+     * Opaque keys are not stored in the handshake's data and it's the user
+     * responsibility to destroy them. Clear ones, instead, are created by
+     * the TLS library and should be destroyed at the same level
+     */
+    if( ! mbedtls_svc_key_id_is_null( handshake->psa_pake_password ) )
```

Comment:
Minor optimisation: we can actually call `destroy()` unconditionally, as if `mbedtls_svc_key_id_is_null()` is true, it will just [do nothing and return success](https://armmbed.github.io/mbed-crypto/html/api/keys/management.html#c.psa_destroy_key). But we should absolutely keep the comment :)